libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -1115,11 +1073,13 @@ void AC_PosControl::run_xy_controller(float dt)
 void AC_PosControl::accel_to_lean_angles(float accel_x_cmss, float accel_y_cmss, float& roll_target, float& pitch_target) const
 {
     float accel_right, accel_forward;
+    float sin_yaw = sinf(_attitude_control.get_att_target_euler().z);
```

Comment:
AHRS goes to a lot of trouble to cache the results... do these update at the same rate that this is being called?  If not, they should be pre-calculated.

Reply:
I think sinf() and cosf() calls are not as expensive as they once were so I think it is probably not worth trying to save CPU by caching the results.  I think at most we would save one sinf() and one cosf() per main loop.

To provide some background, it seems like accel_to_lean_angles is called from 2 places.  AC_Loiter's init_target() which is called when many modes (Land, Loiter, PosHold, etc) are landed.  It is also called when some modes start using the Loiter controller (i9.e. zigzag mode when it switches from autonomous submode to "manual" flight).

It is also called at the end of AC_PosControl::run_xy_controller() which is called at the main loop rate.